### ISSUE TITLE
[Bug] Users and Grants not properly created

### ISSUE THREAD

**Issue by StoffelCPR** [mariadb-operator/mariadb-operator#301]
**Describe the bug**

I'm currently using the opeartor in our cluster using ArgoCD. The opeartor was set up in the namespace `mariadb-operator`.
A mariadb server is setup in a namespace called `dev-team` and the rest of the resource are setup in a namespace called `dev-team-management`.

The first approaches were good and everything went well.. Now I can't get the creations to work.. After several recreations and deletions I got the user back up but the grant won't work even though it's status is created:

```yaml
apiVersion: mariadb.mmontes.io/v1alpha1
kind: Grant
metadata:
  annotations:
    argocd.argoproj.io/tracking-id: >-
      <redacted>
    kubectl.kubernetes.io/last-applied-configuration: >
      <redacted>
  creationTimestamp: '2023-12-12T13:29:42Z'
  finalizers:
    - grant.mariadb.mmontes.io/finalizer
  generation: 1
  labels:
    app.kubernetes.io/instance: dev-team_team-management
  name: item-management-grant
  namespace: dev-team-management
  resourceVersion: '29540079'
  uid: dd88d753-945a-45b8-99fe-1aa1f111535e
spec:
  database: '*'
  grantOption: false
  mariaDbRef:
    name: dev-mariadb-team
    namespace: dev-team
    waitForIt: true
  privileges:
    - CREATE
    - SELECT
    - INSERT
    - UPDATE
  table: '*'
  username: item-management-app
status:
  conditions:
    - lastTransitionTime: '2023-12-12T13:29:42Z'
      message: Created
      reason: Created
      status: 'True'
      type: Ready
```

This is the argocd live manifest with the status added to the running manifest

**Expected behaviour**

I'd expect that the grant is seen in the user after it is created but it isn't.

**Steps to reproduce the bug**

**Additional context**

**Environment details**:
- Kubernetes version: v1.28.3-eks-4f4795d
- mariadb-operator version: 0.22.0
- Install method: helm

**Comment by thecodeassassin** [comment]
+1 same here. Users were not created.

**Comment by pingvinton** [comment]
Hello, I got the same issue. The problem is that the operator doesn't keep track of changes in the password secret
Helm used it before the password secret and the operator worked it out.

I solve the problem when:
```
kubectl delete user <user_name>
kubectl delete grunt <grunt_name> 
```
And apply again.

**Comment by mmontes11** [comment]
Thanks for reporting! 

It is worth mentioning that the SQL resources (users, grants and databases) are no longer reconciled once they are successfully created for the first time in the database (`Ready` condition true). Ideally we should periodically reconcile them to make sure they are recreated if needed, like in this case.

To achieve this, we would by default requeue SQL resources every `30s` to make sure they are in place in the database internal state. This could be customised via the `--requeue-sql` flag available in the controller.

**Comment by mmontes11** [comment]
Please take a look at the following PR:
- https://github.com/mariadb-operator/mariadb-operator/pull/306

It will be released in `v0.0.24`, make sure you upgrade 🚀
